crates/bevy_asset/src/path.rs

```diff
@@ -506,18 +546,21 @@ impl<'a> AssetPath<'a> {
     /// assert!(path.is_unapproved());
     /// ```
     pub fn is_unapproved(&self) -> bool {
-        use std::path::Component;
-        let mut simplified = PathBuf::new();
-        for component in self.path.components() {
+        if self.path.starts_with("/") {
+            return true;
+        }
+        let mut simplified = Vec::new();
+        for component in self.path_components() {
```

Comment:
Would it be more readable and potentially more reusable path parsing if we roll our own version of path components with an enum?

Reply:
it's possible, but I'm not sure about this. Do the other maintainers also agree?

Reply:
I don't have a strong opinion here, leaning to no for simplicity?